src/lib.rs

```diff
@@ -0,0 +1,92 @@
+use lazy_static::lazy_static;
+
+use guest::prelude::*;
+use kubewarden_policy_sdk::wapc_guest as guest;
+
+use k8s_openapi::api::core::v1 as apicore;
+
+extern crate kubewarden_policy_sdk as kubewarden;
+use kubewarden::{logging, protocol_version_guest, request::ValidationRequest, validate_settings};
+
+mod settings;
+use settings::Settings;
+
+use slog::{o, warn, Logger};
+
+lazy_static! {
+    static ref LOG_DRAIN: Logger = Logger::root(
+        logging::KubewardenDrain::new(),
+        o!("policy" => "sample-policy")
+    );
+}
+
+#[no_mangle]
+pub extern "C" fn wapc_init() {
+    register_function("validate", validate);
+    register_function("validate_settings", validate_settings::<Settings>);
+    register_function("protocol_version", protocol_version_guest);
+}
+
+fn validate(payload: &[u8]) -> CallResult {
+    let validation_request: ValidationRequest<Settings> = ValidationRequest::new(payload)?;
+    match validation_request.extract_pod_spec_from_object() {
+        Ok(Some(pod_spec)) => {
+            match validate_pod_priority_class(
+                pod_spec,
+                validation_request.settings.allowed_priority_classes,
+            ) {
+                Ok(_) => kubewarden::accept_request(),
+                Err(err) => kubewarden::reject_request(Some(err.to_owned()), None, None, None),
+            }
+        }
+        Ok(None) => {
+            warn!(LOG_DRAIN, "no PodSpec found");
+            kubewarden::accept_request()
+        }
+        Err(_) => {
+            warn!(LOG_DRAIN, "cannot extract PodSpec");
+            kubewarden::accept_request()
+        }
+    }
+}
+
+#[allow(dead_code)]
```

Comment:
why do we need that?

Reply:
Ops... I added that during some tests during development and forgot to remove it.